### PR TITLE
Add will display cell signal to CollectionViewDelegate

### PR DESCRIPTION
**Pull request by Iamsamwhoami**
This PR adds a signal for `willDisplayCell`.

Can be used like this:

```swift
bag += collectionKit.delegate.willDisplayCell.onValue { cell in
  // do something exciting with cell
}

### CODE REVIEW

**Comment by mansbernhardt** [comment]
It's great that you started to fill in the missing gaps in our delegates. We discussed that we need to address that in https://github.com/iZettle/Form/pull/37#issuecomment-423896975.

But, as non-body has come around to complete the support for our table and collection delegate/data-source methods, I wonder if that is something you would like to address?

FYI @fennecos

**Comment by Iamsamwhoami** [comment]
@mansbernhardt yeah, I found some things to be missing, so I've just implemented those that I really needed for now.

I can absolutely take a stab at implementing the rest of the delegate methods (atleast for collection views), but I'll probably not get around to it until after Christmas 😄

File: Form/CollectionViewDelegate.swift [hunk 1]
```diff
@@ -23,6 +23,7 @@ public final class CollectionViewDelegate<Section, Row>: ScrollViewDelegate, UIC
     private let didSelectCallbacker = Callbacker<TableIndex>()
     private let didEndDisplayingCellCallbacker = Callbacker<UICollectionViewCell>()
     private let didEndDisplayingSupplementaryViewCallbacker = Callbacker<(kind: String, view: UICollectionReusableView)>()
+    private let willDisplayCellCallbacker = Callbacker<UICollectionViewCell>()
```
Comment (mansbernhardt) [inline Form/CollectionViewDelegate.swift]: I think we should forward all available information so this should rather be:

```swift
private let willDisplayCellCallbacker = Callbacker<(UICollectionViewCell, TableIndex)>()
```

See corresponding method in `TableViewDelegate` 

```swift
    private let willDisplayCellCallbacker = Callbacker<(UITableViewCell, TableIndex)>()
```

Comment (Iamsamwhoami) [reply Form/CollectionViewDelegate.swift]: @mansbernhardt sorted! 👌

**Comment by Iamsamwhoami** [comment]
@mansbernhardt merged this now  🙌 I am a bit swamped right now, but will be sure to revisit this as soon as I can :D

**Pull request merged**